app/cdap/components/Connections/Browser/GenericBrowser/index.tsx

```diff
@@ -168,42 +170,52 @@ export function GenericBrowser({ selectedConnection }) {
           <SearchField onChange={handleSearchChange} value={searchStringDisplay} />
         </div>
       </div>
-      <Table columnTemplate={columnTemplate} classes={{ grid: classes.grid }}>
-        <TableHeader>
-          <TableRow>
-            {headers.map((header) => {
-              return <TableCell key={header}>{header}</TableCell>;
+      <If condition={!isEmpty && !error}>
+        <Table columnTemplate={columnTemplate} classes={{ grid: classes.grid }}>
+          <TableHeader>
+            <TableRow>
+              {headers.map((header) => {
+                return <TableCell key={header}>{header}</TableCell>;
+              })}
+            </TableRow>
+          </TableHeader>
+          <TableBody>
+            {filteredEntities.map((entity, i) => {
+              return (
+                <TableRow
+                  key={i}
+                  to={`/ns/${getCurrentNamespace()}/connections/${selectedConnection}?path=${getPath(
+                    entity.name
+                  )}`}
+                  onClick={() => onExplore(entity.name)}
+                >
+                  <TableCell>
+                    <div className={classes.nameWrapper}>
+                      <If condition={ICON_MAP[entity.type]}>{ICON_MAP[entity.type]}</If>
+                      <div>{entity.name}</div>
+                    </div>
+                  </TableCell>
+                  <TableCell>{entity.type}</TableCell>
+                  {entity.properties.map((prop) => {
+                    if (prop.type === 'Timestamp') {
```

Comment:
Nit: this could be a ternary to avoid use of imperative code inside an element tree expression. though I see you didn't add this code newly, I figure I might as well comment on meaning-per-token improvements while it's affected by a PR! up to you.

Reply:
Cleaned this up a bit.